### ISSUE TITLE
Docker build issue with r1.14 branch

### ISSUE THREAD

**Issue by karthikvadla** [tensorflow/serving#1363]
Hi, i'm trying to build tensorflow serving image with `r1.14` branch. I see below parse issue with the branch
```
Step 21/24 : RUN bazel build --color=yes --curses=yes     ${TF_SERVING_BAZEL_OPTIONS}     --verbose_failures     --output_filter=DONT_MATCH_ANYTHING     ${TF_SERVING_BUILD_OPTIONS}     tensorflow_serving/tools/pip_package:build_pip_package &&     bazel-bin/tensorflow_serving/tools/pip_package/build_pip_package     /tmp/pip &&     pip --no-cache-dir install --upgrade     /tmp/pip/tensorflow_serving_api-*.whl &&     rm -rf /tmp/pip
 ---> Running in 486382c275ec
Starting local Bazel server and connecting to it...
INFO: Analysed target //tensorflow_serving/tools/pip_package:build_pip_package (26 packages loaded, 798 targets configured).
INFO: Found 1 target...
Target //tensorflow_serving/tools/pip_package:build_pip_package up-to-date:
  bazel-bin/tensorflow_serving/tools/pip_package/build_pip_package
INFO: Elapsed time: 5.082s, Critical Path: 0.58s
INFO: 83 processes: 83 local.
INFO: Build completed successfully, 87 total actions
Wed May 29 20:42:56 UTC 2019 : === Using tmpdir: /tmp/tmp.SM4La0lE8S
Adding python files
Adding package setup files
/tmp/tmp.SM4La0lE8S /tensorflow-serving
Wed May 29 20:42:56 UTC 2019 : === Building wheel (CPU)
error in tensorflow-serving-api setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Invalid requirement, parse error at "'=1.14.0r'"
The command '/bin/sh -c bazel build --color=yes --curses=yes     ${TF_SERVING_BAZEL_OPTIONS}     --verbose_failures     --output_filter=DONT_MATCH_ANYTHING     ${TF_SERVING_BUILD_OPTIONS}     tensorflow_serving/tools/pip_package:build_pip_package &&     bazel-bin/tensorflow_serving/tools/pip_package/build_pip_package     /tmp/pip &&     pip --no-cache-dir install --upgrade     /tmp/pip/tensorflow_serving_api-*.whl &&     rm -rf /tmp/pip' returned a non-zero code: 1
```

I used docker build command as below

```
docker build \
    --build-arg TF_SERVING_VERSION_GIT_BRANCH="r1.14" \
    --build-arg HTTP_PROXY=${HTTP_PROXY} \
    --build-arg HTTPS_PROXY=${HTTPS_PROXY} \
    --build-arg http_proxy=${http_proxy} \
    --build-arg https_proxy=${https_proxy} \
    -f Dockerfile.devel -t tensorflow/serving:latest-devel .
```
Any help please.

**Comment by karthikvadla** [comment]
@lilao can't we build images based of branches? just wonder how it's working with `master`. 🤔

**Comment by karthikvadla** [comment]
@gautamvasudevan This is different question, where is tensorflow version set in tensorflow serving?
I know when i set `1.13.0` as `TF_SERVING_VERSION_GIT_BRANCH` it uses same version of tensorflow. Do we have a way to control different version of tensorflow? or any pointers to code base where this is configured automatically?

**Comment by ashahba** [comment]
@karthikvadla Thanks for filing this issue.
In the meantime I created a PR here: #1367 to unblock you.
Here is how you use it:

```
git clone git@github.com:karthikvadla/serving.git
git remote add upstream git@github.com:tensorflow/serving.git
git remote add ashahba git@github.com:ashahba/serving.git
```
Now checkout my branch:
```
git remote update --prune
git checkout ashahba ashahba/future
```
and run your build command as usual:
```
docker build --build-arg TF_SERVING_VERSION_GIT_BRANCH="master"  -f Dockerfile.devel -t tensorflow/serving:latest-devel .
```

**Comment by karthikvadla** [comment]
@ashahba Thank you . Will try that now.

**Comment by karthikvadla** [comment]
https://github.com/tensorflow/serving/pull/1367#discussion_r289582625 Please check here. Your patch fixed different error i have with master branch. Still i see parse issue with `r1.14` branch.

**Comment by ashahba** [comment]
@karthikvadla `r1.14` should be good now, since `r.1.14` branch in `tensorflow` was patched earlier today here: https://github.com/tensorflow/tensorflow/pull/29233
Would you please try building the container with `r1.14` branch and let us know?

Thanks.

**Comment by gowtham-kp** [comment]
Hey @karthikvadla, Are you still facing this issue? Can we close this if it is resolved on your end? Thanks!

**Comment by netfs** [comment]
fwiw our official docker builds are out for r1.14 (rc0) on hub.docker.com. so this is working now.

**Comment by netfs** [comment]
sorry did not want to close the issue, just comment.

**Comment by karthikvadla** [comment]
I will test and update you.

**Comment by karthikvadla** [comment]
```
docker build \
--build-arg TF_SERVING_VERSION_GIT_BRANCH="r1.14" \
--build-arg HTTP_PROXY=${HTTP_PROXY} \
--build-arg HTTPS_PROXY=${HTTPS_PROXY} \
--build-arg http_proxy=${http_proxy} \
--build-arg https_proxy=${https_proxy} \
-f Dockerfile.devel-mkl -t tensorflow/serving:latest-devel-mkl .
```

```
/tensorflow-serving
Tue Jun 4 20:47:00 UTC 2019 : === Output wheel file is in: /tmp/pip
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Processing /tmp/pip/tensorflow_serving_api-1.14.0rc0-py2.py3-none-any.whl
ERROR: Exception:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/pip/_internal/cli/base_command.py", line 178, in main
    status = self.run(options, args)
  File "/usr/local/lib/python2.7/dist-packages/pip/_internal/commands/install.py", line 352, in run
    resolver.resolve(requirement_set)
  File "/usr/local/lib/python2.7/dist-packages/pip/_internal/resolve.py", line 131, in resolve
    self._resolve_one(requirement_set, req)
  File "/usr/local/lib/python2.7/dist-packages/pip/_internal/resolve.py", line 345, in _resolve_one
    set(req_to_install.extras) - set(dist.extras)
  File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 2969, in extras
    return [dep for dep in self._dep_map if dep]
  File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 3014, in _dep_map
    self.__dep_map = self._compute_dependencies()
  File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 3024, in _compute_dependencies
    reqs.extend(parse_requirements(req))
  File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 3085, in parse_requirements
    yield Requirement(line)
  File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 3094, in __init__
    raise RequirementParseError(str(e))
RequirementParseError: Parse error at "'(===1.14'": Expected stringEnd
The command '/bin/sh -c bazel build --color=yes --curses=yes     ${TF_SERVING_BAZEL_OPTIONS}     --verbose_failures     --output_filter=DONT_MATCH_ANYTHING     ${TF_SERVING_BUILD_OPTIONS}     tensorflow_serving/tools/pip_package:build_pip_package &&     bazel-bin/tensorflow_serving/tools/pip_package/build_pip_package     /tmp/pip &&     pip --no-cache-dir install --upgrade /tmp/pip/tensorflow_serving*.whl &&     rm -rf /tmp/pip' returned a non-zero code: 2
```

Am i doing anything wrong?

**Comment by netfs** [comment]
> ```
> docker build \
> --build-arg TF_SERVING_VERSION_GIT_BRANCH="r1.14" \
> --build-arg HTTP_PROXY=${HTTP_PROXY} \
> --build-arg HTTPS_PROXY=${HTTPS_PROXY} \
> --build-arg http_proxy=${http_proxy} \
> --build-arg https_proxy=${https_proxy} \
> -f Dockerfile.devel-mkl -t tensorflow/serving:latest-devel-mkl .
> ```
> 
> ```
> /tensorflow-serving
> Tue Jun 4 20:47:00 UTC 2019 : === Output wheel file is in: /tmp/pip
> DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
> Processing /tmp/pip/tensorflow_serving_api-1.14.0rc0-py2.py3-none-any.whl
> ERROR: Exception:
> Traceback (most recent call last):
>   File "/usr/local/lib/python2.7/dist-packages/pip/_internal/cli/base_command.py", line 178, in main
>     status = self.run(options, args)
>   File "/usr/local/lib/python2.7/dist-packages/pip/_internal/commands/install.py", line 352, in run
>     resolver.resolve(requirement_set)
>   File "/usr/local/lib/python2.7/dist-packages/pip/_internal/resolve.py", line 131, in resolve
>     self._resolve_one(requirement_set, req)
>   File "/usr/local/lib/python2.7/dist-packages/pip/_internal/resolve.py", line 345, in _resolve_one
>     set(req_to_install.extras) - set(dist.extras)
>   File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 2969, in extras
>     return [dep for dep in self._dep_map if dep]
>   File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 3014, in _dep_map
>     self.__dep_map = self._compute_dependencies()
>   File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 3024, in _compute_dependencies
>     reqs.extend(parse_requirements(req))
>   File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 3085, in parse_requirements
>     yield Requirement(line)
>   File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 3094, in __init__
>     raise RequirementParseError(str(e))
> RequirementParseError: Parse error at "'(===1.14'": Expected stringEnd
> The command '/bin/sh -c bazel build --color=yes --curses=yes     ${TF_SERVING_BAZEL_OPTIONS}     --verbose_failures     --output_filter=DONT_MATCH_ANYTHING     ${TF_SERVING_BUILD_OPTIONS}     tensorflow_serving/tools/pip_package:build_pip_package &&     bazel-bin/tensorflow_serving/tools/pip_package/build_pip_package     /tmp/pip &&     pip --no-cache-dir install --upgrade /tmp/pip/tensorflow_serving*.whl &&     rm -rf /tmp/pip' returned a non-zero code: 2
> ```
> 
> Am i doing anything wrong?

are you using the latest r1.14 sources? the `===` indicates you are not.

**Comment by karthikvadla** [comment]
I didn't get this `latest r1.14 sources` 🤔 , what do you mean by this?

I did as below.
```
git clone git@github.com:tensorflow/serving
cd serving/tensorflow_serving/tools/docker
```
and then ran `docker build` command.

**Comment by netfs** [comment]
the error on `===` should be fixed in 1.14 branch by https://github.com/tensorflow/serving/commit/91484e1a7087744dc7b0c5ff773b84d58dbab0d5

so its strange that your build is seeing `===` in r1.14 sources.

**Comment by karthikvadla** [comment]
Oh yeah i cleaned up all my docker cache and re-built with `r1.14` branch. Issue resolved. Closing it now.
